ledger/light/handle/set_outgoing_request.go

```diff
@@ -42,20 +42,25 @@ func NewSetOutgoingRequest(dep *proc.Dependencies, msg payload.Meta, passed bool
 }
 
 func (s *SetOutgoingRequest) Present(ctx context.Context, f flow.Flow) error {
-	msg := payload.SetOutgoingRequest{}
-	err := msg.Unmarshal(s.message.Payload)
+	pl, err := payload.Unmarshal(s.message.Payload)
 	if err != nil {
-		return errors.Wrap(err, "SetOutgoingRequest.Present: failed to unmarshal SetIncomingRequest message")
+		return errors.Wrap(err, "failed to unmarshal SetOutgoingRequest message")
+	}
+	msg, ok := pl.(*payload.SetOutgoingRequest)
+	if !ok {
+		return fmt.Errorf("wrong request type: %T", pl)
 	}
 
-	rec := record.Unwrap(&msg.Request)
+	virtual := msg.Request
+	rec := record.Unwrap(&virtual)
 	request, ok := rec.(*record.OutgoingRequest)
 	if !ok {
-		return fmt.Errorf("SetOutgoingRequest.Present: wrong request type: %T", rec)
+		return fmt.Errorf("wrong request type: %T", rec)
 	}
 
-	if request.AffinityRef() == nil {
-		return errors.New("SetOutgoingRequest.Present: object is nil")
+	objectID := *request.AffinityRef().Record()
```

Comment:
nil pointer dereference. Better use `request.AffinityRef() == nil || *request.AffinityRef().Record().IsEmpty()`

Reply:
We have non-reference field Caller in OutgoingRequest struct (Caller is returned when we call AffinityRef method on request). So we cant't get nil reference here.